### PR TITLE
Fix an issue with Java equality of DLN

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/2316

The following returns `true` as expected:
```Java
new DLN("1.1.1").equals(new DLN("1.1.1"))
```

However, the following unexpectedly returns `false`, when one would expect it to return `true`:
```java
Optional.of(new DLN("1.1.1").equals(Optional.of(new DLN("1.1.1"))))
```

This is due to a strange definition (and implementation) of `equals` in `NodeId`; I have removed this potentially dangerous pattern so that you get expected behaviour.

Most of the test I adapted from @joewiz. I would be happy if Joe could check he agrees with them still ;-)

### CODE REVIEW

File: exist-core/src/test/xquery/util/get-fragment-between.xqm [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+declare
+    %test:tearDown
+function gfb:teardown() {
+    xmldb:remove("/db/test-gfb")
+};
+
+declare
+    %test:assertEquals('<pb facs="1.jpg"></pb><p>Aus dem Leben einer Kartoffel.</p>')
```
Comment (joewiz) [inline exist-core/src/test/xquery/util/get-fragment-between.xqm]: The fact that the function's 4th parameter is `true()` means the resulting fragment should preserve the root namespace:

> The fourth argument, `$display-root-namespace`, is a boolean value for displaying the root node namespace. If it is set to `true()` the attribute `xmlns` in the root node of the result sequence is determined explicitly from the `$beginning-node`.

```suggestion
    %test:assertEquals('<pb xmlns="http://www.tei-c.org/ns/1.0" facs="1.jpg"></pb><p>Aus dem Leben einer Kartoffel.</p>')
```

**Comment by joewiz** [comment]
Restarted the one failing Travis CI build. Still awaiting final Appveyor build.

File: exist-core/src/test/xquery/util/get-fragment-between.xqm (hunk 1 above)
Comment (adamretter) [reply exist-core/src/test/xquery/util/get-fragment-between.xqm]: I think the 4th parameter only has an effect when the 3rd parameter is `true`. I have committed two further tests to try and illustrate this.

Unfortunately it is very hard to know what the 'correct' behaviour of `util:get-fragment-between` is exactly mean to be as there is no documentation :-/

Comment (joewiz) [reply exist-core/src/test/xquery/util/get-fragment-between.xqm]: Good point about the thin documentation for this function (explaining why I wasn’t sure whether the second node was preserved or discarded). 

But is it necessary to limit namespace preservation to when a root node is added? Surely users would need a way to preserve the namespace of the resulting nodes. 

My suggestion also neglected to include the namespace on the 2nd node.

Comment (adamretter) [reply exist-core/src/test/xquery/util/get-fragment-between.xqm]: > But is it necessary to limit namespace preservation to when a root node is added? Surely users would need a way to preserve the namespace of the resulting nodes.

No... but my goal was to fix the regression, and just get it working as it was in eXist-db 4.3.1.

**Comment by joewiz** [comment]
Fair enough! Thank you!!!

**Pull request merged**